### PR TITLE
Various layout tweaks

### PR DESCRIPTION
**Pull request by Herrie82**
This should fix the S and XS behavior for tablet layouts (more similar to legacy) + further enhance the phone layouts as well.

### CODE REVIEW

File: plugins/en/qml/Keyboard_en_phone_email.qml [hunk 1]
```diff
@@ -35,66 +35,68 @@ KeyPad {
...
+
+            ShiftKey {id: shiftKey; anchors.left: parent.left; }
+            CharKey { id: zKey; anchors.left: shiftKey.right; width: UI.keyWidth * 8/7; label: "z"; shifted: "Z"; extended: ["z", "ž", "ź", "ż"]; extendedShifted: ["Z", "Ž", "Ź", "Ż"]; }
+            CharKey { id: xKey; anchors.left: zKey.right; width: UI.keyWidth * 8/7; label: "x"; shifted: "X"; extended: ["x", "Rec", "Mute"]; extendedShifted: ["X", "Rec", "Mute"]; }
+            CharKey { id: cKey; anchors.left: xKey.right; width: UI.keyWidth * 8/7; label: "c"; shifted: "C"; extended: ["c", "ç", "ć", "©", "¢"]; extendedShifted: ["C", "Ç", "Ć", "©", "¢"]; }
+            CharKey { id: vKey; anchors.left: cKey.right; width: UI.keyWidth * 8/7; label: "v"; shifted: "V"; }
+            CharKey { id: bKey; anchors.left: vKey.right; width: UI.keyWidth * 8/7; label: "b"; shifted: "B"; }
+            CharKey { id: nKey; anchors.left: bKey.right; width: UI.keyWidth * 8/7; label: "n"; shifted: "N"; extended: ["n", "ñ", "ń"]; extendedShifted: ["N", "Ñ", "Ń"]; }
+            CharKey { id: mKey; anchors.left: nKey.right; width: UI.keyWidth * 8/7; label: "m"; shifted: "M"; extended: ["m", "μ"]; extendedShifted: ["M", "Μ"]; }
```
Comment (Tofee) [inline plugins/en/qml/Keyboard_en_phone_email.qml]: those are regular keys, is it intended that they have a different width?

File: plugins/en/qml/Keyboard_en_tablet.qml [hunk 2]
```diff
@@ -36,7 +36,7 @@ KeyPad {
             anchors.horizontalCenter: parent.horizontalCenter;
             spacing: 0
 
-            height: keyHeight * 0.74
+            height: UI.keyboardSizeChoice === "XS" ? keyHeight * 0.82 : keyHeight * 0.74
```
Comment (Tofee) [inline plugins/en/qml/Keyboard_en_tablet.qml]: only for the regular tablet layout, not the "email" or "url" ?

File: qml/keys/AnnotatedKey.qml [hunk 3]
```diff
@@ -128,42 +129,31 @@ Item {
...
-        anchors.leftMargin: useHorizontalLayout ? Units.gu(2.0) : 0
-
-        anchors.horizontalCenter: useHorizontalLayout ? undefined : buttonImage.horizontalCenter
-        anchors.horizontalCenterOffset: useHorizontalLayout ? Units.gu(2.75) : 0
+        anchors.horizontalCenter: buttonImage.horizontalCenter
+        anchors.horizontalCenterOffset: useHorizontalLayout ? UI.keyWidth/-6 : 0
 
         anchors.verticalCenter: parent.verticalCenter
-        anchors.verticalCenterOffset: useHorizontalLayout ? Units.gu(-0.25) : Units.gu(-1.75)
+        anchors.verticalCenterOffset: useHorizontalLayout ? Units.gu(-0.25) : UI.formFactor === "phone" && UI.isLandscape ? Units.gu (-1.00) : Units.gu(-1.75)
```
Comment (Tofee) [inline qml/keys/AnnotatedKey.qml]: maybe we should use keyHeight for that sort of precise layout ? like you used keyWidth just above, which is pretty smart imho

File: qml/keys/ExtendedListSelector.qml [hunk 4]
```diff
@@ -70,9 +70,9 @@ Item {
         anchors.bottom: anchorItem.bottom
         anchors.bottomMargin: -8
         x: isOnLeftSideOfScreen ? (anchorItem.x) : (anchorItem.x+anchorItem.width-popoverBackground.width)
-        width: Math.max(UI.keyWidth, rowOfKeys.width + 10*2)
-        height: ((30 + numberOfLines * 60))
-
+        width: Math.max(Units.gu(8.0) * (keyRepeater.count <=4 ? keyRepeater.count : 4), rowOfKeys.width + 10*2)
+        height: ((Units.gu(3.0) + numberOfLines * Units.gu(6.3)))
```
Comment (Tofee) [inline qml/keys/ExtendedListSelector.qml]: using popover.keyHeight and popover.heyWidth here would be better

File: qml/keys/ExtendedListSelector.qml [hunk 5]
```diff
@@ -150,8 +150,8 @@ Item {
                     anchors.centerIn: parent
                     id: popupKeyImage
                     source: UI.imagePopupKey
-                    width: 60
-                    height: 60
+                    width: Units.gu(8.0) 
+                    height: Units.gu(6.3)
```
Comment (Tofee) [inline qml/keys/ExtendedListSelector.qml]: using popover.keyHeight and popover.heyWidth here would be better

File: qml/keys/ExtendedListSelector.qml [hunk 6]
```diff
@@ -131,7 +131,7 @@ Item {
         id: rowOfKeys
         anchors.centerIn: popoverBackground
         anchors.verticalCenterOffset: -5
-        width: Math.min(keyRepeater.count, 4) * 60
+        width: Math.min(keyRepeater.count, 4) * Units.gu(8.0)
```
Comment (Tofee) [inline qml/keys/ExtendedListSelector.qml]: using popover.keyHeight here would be better

File: qml/keys/SymbolShiftKey.qml [hunk 7]
```diff
@@ -26,7 +26,7 @@ ActionKey {
     shifted: "+=[]";
     action: "symbols";
 
-    fontSize: UI.symbolShiftKeyFontSize;
+    fontSize: UI.xsFontSize;
```
Comment (Tofee) [inline qml/keys/SymbolShiftKey.qml]: it's a bit weird to not use UI.symbolShiftKeyFontSize for the SymbolShiftKey font size... Is UI.symbolShiftKeyFontSize used elsewhere now ? Shouldn't we prefer simply changing symbolShiftKeyFontSize's value ?

Comment (Herrie82) [inline qml/keys/SymbolShiftKey.qml]: Removed symbilkShiftKeyFontSize altogether now :)

File: plugins/en/qml/Keyboard_en_tablet.qml (hunk 2 above)
Comment (Herrie82) [inline plugins/en/qml/Keyboard_en_tablet.qml]: Added those as well now, well spotted :)

File: plugins/en/qml/Keyboard_en_phone_email.qml (hunk 1 above)
Comment (Herrie82) [inline plugins/en/qml/Keyboard_en_phone_email.qml]: It's a bit of nasty hack. OWO VKB for phone seems to use different keywidths for different rows, while we take it based on the longest row it seems. This dirty hack fixes the alignment of the keys by making them wider to look similar to OWO.

**Pull request merged**